bundles/org.openhab.binding.ecowatt/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -34,12 +34,22 @@
 			</channel>
 			<channel id="currentHourSignal" typeId="signal">
 				<label>Current Hour Signal</label>
-				<description>The signal relating to the forecast consumption level for the current hour. Values are 1 for normal
-					consumption (green), 2 for strained electrical system (orange) and 3 for very strained electrical system (red).</description>
+				<description>The signal relating to the forecast consumption level for the current hour. Values are 0 for normal
+					consumption (green) and carbon-free production, 1 for normal consumption (green), 2 for strained electrical system
+					(orange) and 3 for very strained electrical system (red).</description>
 			</channel>
 		</channels>
 
 		<config-description>
+			<parameter name="apiVersion" type="integer" required="false">
+				<label>API version</label>
+				<description>The version of the Ecowatt tile to which you subscribed in the RTE portal.</description>
+				<options>
+					<option value="4">V4.0 (deprecated)</option>
+					<option value="5">V5.0</option>
+				</options>
+				<default>4</default>
```

Comment:
I understand it will be a breaking change, but nevertheless, shouldn't we default the current version rather than the deprecated version? Otherwise new installations will start off by using the wrong version. A breaking note could be added advising existing users to either set API version 4 (quick short-term solution) or upgrade to API v5.

Reply:
The switch to version 5 is not ultra simple and using version 4 is still working. For all users having a working setup, I don't want them to have the binding failing when upgrading to OH 4.1 because the binding is now using API version 5 by default. Take in mind that this parameter was not existing when the user created its thing.
New users will of course see this parameter and will set it to the proper version.

Reply:
> Otherwise new installations will start off by using the wrong version

No because they will create the application in the RTE portal just before and see what the version they subscribed.
Additionally, when they will see "V4.0 (deprecated)", they should probably switch to V5.0.
And finally a new user should normally have a quick look to the binding README and so should see the explanation of this API version. An old user will probably not go back to hte README.

Reply:
The only issue I see with this approach is that this situation will persist, i.e. v4 will still have to be the default when v6 and v7 are added, as otherwise you'll just postpone the breaking change. If you have considered this already, I think we can merge since I agree there is no really good way to solve this, at least nothing that comes to my mind right now.

Reply:
What would happen if you keep 4 as default value in `EcowattConfiguration`, but bump to 5 as default in thing definition XML?

Reply:
> The only issue I see with this approach is that this situation will persist, i.e. v4 will still have to be the default when v6 and v7 are added, as otherwise you'll just postpone the breaking change.

I will probably change the default and remove v4 when it will be reported to be stopped.

> What would happen if you keep 4 as default value in EcowattConfiguration, but bump to 5 as default here in the thing definition XML?

Normally in that case I should get 5 in the Java object when initializing the thing.

Reply:
> > What would happen if you keep 4 as default value in EcowattConfiguration, but bump to 5 as default here in the thing definition XML?
> 
> Normally in that case I should get 5 in the Java object when initializing the thing.

I need to test this, but I believe if the value is missing in the configuration, you will get the defaulted value in the DTO, i.e. 4.